llvm/utils/TableGen/AsmMatcherEmitter.cpp

```diff
@@ -3253,7 +3276,7 @@ void AsmMatcherEmitter::run(raw_ostream &OS) {
         const MatchableInfo &A = **I;
         const MatchableInfo &B = **J;
 
-        if (A.couldMatchAmbiguouslyWith(B)) {
+        if (A.couldMatchAmbiguouslyWith(B, Target)) {
```

Comment:
```suggestion
        bool PreferSmallerInstructions =
            AsmParser->getValueAsBit("PreferSmallerInstructions");
        if (A.couldMatchAmbiguouslyWith(B, PreferSmallerInstructions)) {
```
Same for shouldBeMatchedBefore above.

Reply:
Ah yes, I agree this is a good idea. I will address in https://github.com/llvm/llvm-project/pull/83587.